src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -698,33 +698,31 @@ private ProcessThreadTimes GetProcessTimes()
             }
         }
 
-        private static void SetPrivilege(string privilegeName, int attrib)
+        private static unsafe void SetPrivilege(string privilegeName, int attrib)
         {
-            SafeTokenHandle hToken = null;
-            Interop.Advapi32.LUID debugValue = new Interop.Advapi32.LUID();
-
             // this is only a "pseudo handle" to the current process - no need to close it later
             SafeProcessHandle processHandle = Interop.Kernel32.GetCurrentProcess();
 
             // get the process token so we can adjust the privilege on it.  We DO need to
             // close the token when we're done with it.
-            if (!Interop.Advapi32.OpenProcessToken(processHandle, Interop.Kernel32.HandleOptions.TOKEN_ADJUST_PRIVILEGES, out hToken))
+            if (!Interop.Advapi32.OpenProcessToken(processHandle, Interop.Kernel32.HandleOptions.TOKEN_ADJUST_PRIVILEGES, out SafeTokenHandle hToken))
             {
```

Comment:
Consider calling hToken.Dispose() here to avoid it needing to be finalized.

Reply:
I don't understand. This isn't even in a try.

Reply:
To be clear, I was talking about the case where this fails and throws an exception, in which case the invalid safe handle will be left for finalization. It's of course not a big deal.